erigon-lib/downloader/downloader.go

```diff
@@ -824,6 +835,19 @@ func (d *Downloader) mainLoop(silent bool) error {
 
 			for _, t := range d.torrentClient.Torrents() {
 				if urls, ok := d.webseeds.ByFileName(t.Name()); ok {
+					// if we have created a torrent bit it has no info assume that the
+					// webseed download either has not been called yet or has failed and
+					// try again here - otherwise the torrent will be left with no info
+					if t.Info() == nil {
```

Comment:
```
select {
case <-t.GotInfo():
   continue
default:
}
```
is better than ` t.Info()` from "mutex contention" perspective.

Reply:
Info now has its own mutex - so contention is minimal - as its only locked written by set info.  So I think that t.Info() should be preferred for call simplicity.